From e29dded5ed13f86d610ac11172f7bac6552bfd0f Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair@alistair23.me>
Date: Tue, 27 Jan 2026 21:57:57 +1000
Subject: [PATCH] telephonySupport: Don't delete other Bearers

Sparebar incorrectly deletes other Bearers which breaks 81voltd or data
connections. Spacebar shouldn't be deleting Bearers so just remove the
logic.

Resolved: https://invent.kde.org/plasma-mobile/spacebar/-/issues/71
Signed-off-by: Alistair Francis <alistair@alistair23.me>
---
 lib/telephonySupport/modemcontroller.cpp | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/lib/telephonySupport/modemcontroller.cpp b/lib/telephonySupport/modemcontroller.cpp
index eba5f01..18c9127 100644
--- a/lib/telephonySupport/modemcontroller.cpp
+++ b/lib/telephonySupport/modemcontroller.cpp
@@ -65,14 +65,7 @@ void ModemController::init(std::optional<QString> modemPath)
 
     connect(m_interface.get(), &ModemManager::Modem::bearerAdded, this, [this](const QString &bearer) {
         m_bearer = m_interface->findBearer(bearer);
-
         QList<QSharedPointer<ModemManager::Bearer>> bearers = m_interface->listBearers();
-        for (const ModemManager::Bearer::Ptr &item : bearers) {
-            if (item->uni() != bearer) {
-                item->disconnect();
-                m_interface->deleteBearer(item->uni());
-            }
-        }
 
         // whether or not the bearer is connected and thus whether packet data communication using this bearer is possible
         connect(m_bearer.data(), &ModemManager::Bearer::connectedChanged, this, [this](bool isConnected) {
-- 
2.52.0

